mvn-extension/src/main/java/org/arquillian/smart/testing/mvn/ext/SmartTestingMavenConfigurer.java

```diff
@@ -88,7 +100,62 @@ private void calculateChanges() {
 
     private void configureExtension(MavenSession session, Configuration configuration) {
         final MavenProjectConfigurator mavenProjectConfigurator = new MavenProjectConfigurator(configuration);
-        session.getAllProjects().forEach(mavenProject -> mavenProjectConfigurator.configureTestRunner(mavenProject.getModel()));
+        session.getAllProjects().forEach(mavenProject -> {
+            mavenProjectConfigurator.configureTestRunner(mavenProject.getModel());
+            if (isFailedStrategyUsed()) {
+                copySurefireReports(mavenProject.getModel());
+            }
+        });
+    }
+
+    private void purgeReports(MavenSession session) {
```

Comment:
Ultimately let's move this logic to its own class.

Reply:
Sure, I'll move all the logic related to surefire reports to a separated class `SurefireReportStorage` - I didn't make it on Friday :-)